### PR TITLE
Update the progress status with the ticker

### PR DESCRIPTION
**Pull request by bchapuis**
Apparently, the progress reporter already uses a ticker initialized at 1second, however, the updateProgress function is called every time new stats are added. I just tried to update the progress with the ticker only (60fps), to see if it has an effect on cpu usage on mac OS X. I may miss some points but it doesn't looks like the root cause of issue #263.

### CODE REVIEW

**Comment by fd0** [comment]
Thanks for trying to fix this! Unfortunately, this isn't a good fix. It updates the status 60 times per second although there may be nothing to update. I'd suggest leaving the ticker at a second (so that the progress is updated at least once per second). Then add a new field `lastUpdate time.Time` to the `Progress` struct, check this in `Report()` and only call `updateProgress()` when since the last update more than 16ms have passed. e.g.:

```go
func (p *Progress) Report(s Stat) {
	[...]
	needUpdate := false

	p.curM.Lock()
	p.cur.Add(s)
	cur := p.cur
	if time.Since(p.lastUpdate) > 16 * time.Millisecond {
		p.lastUpdate = time.Now()
		needUpdate = true
	}
	p.curM.Unlock()

	if needUpdate {
		p.updateProgress(cur, false)
	}
}
```

**Comment by bchapuis** [comment]
Ok, your code worked as it is :)

**Comment by fd0** [comment]
Cool, I didn't expect that. I'll merge this later today. If you have the time, you could squash both commits.

Maybe @rubenv can give this code a try.

**Comment by rubenv** [comment]
Looks like a perfect fix. CPU usage is back where it should be: within restic.

LGTM

File: progress.go [hunk 1]
```diff
@@ -92,9 +95,17 @@ func (p *Progress) Report(s Stat) {
 	p.curM.Lock()
 	p.cur.Add(s)
 	cur := p.cur
+	needUpdate := false
+	if time.Since(p.lastUpdate) > minTickerTime {
```
Comment (fw42) [inline progress.go]: how about extracting this into a new method?
```
func (p *Progress) needUpdate() { ... }
```
I think that would improve readability. And then use `if p.needUpdate() { ... }` below.

Comment (fw42) [inline progress.go]: also right now this code that you are adding is mutexed by `curM`. Is that intentional/necessary? I don't see why.

Comment (fd0) [inline progress.go]: That is intentional. `Report()` is called from multiple goroutines all read (and update) `p.lastUpdate`.

**Pull request merged**